File: segundo/1er cuatri/PDOO/irrgarten_entrega1/Irrgarten_java/src/irrgarten/Shield.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author alvarogz
 */
public class Shield {
    private float protection;
    private int uses;
    
    public Shield(float protection, int uses) {
        this.protection = protection;
        this.uses = uses;
    }
    
    public float protect() {
        if (this.uses > 0){
            this.uses--;
            return this.protection;
        }
        else
            return 0;
    }
    
    @Override
    public String toString() {
        return "S[" + this.protection + ", " + this.uses + "]";
    }
    
    public boolean discard() {
        return Dice.discardElement(uses);
    }
}

File: segundo/1er cuatri/PDOO/irrgarten_entrega1/irrgarten_ruby/Irrgarten/shield.rb
# frozen_string_literal: true
require_relative 'dice'

class Shield
  def initialize(protection, uses)
    @protection = protection
    @uses = uses
  end

  def protect
    if @uses > 0
      @uses = @uses-1
      @protection
    else
      0.to_f
    end
  end

  def to_s
    "S[#{@protection}, #{@uses}]"
  end

  def inspect
    to_s
  end

  def discard
    Dice.discard_element(@uses)
  end
end
